src/renderers/shaders/ShaderChunk/tonemapping_pars_fragment.glsl.js

```diff
@@ -153,6 +158,9 @@ vec3 AgXToneMapping( vec3 color ) {
 
 	color = LINEAR_REC2020_TO_LINEAR_SRGB * color;
 
+	// Gamut mapping. Simple clamp for now.
+	color = clamp( color, 0.0, 1.0 );
+
```

Comment:
This isn't done in our other color conversion functions (see [P3 to Linear sRGB conversion](https://github.com/mrdoob/three.js/blob/4acb78002425969f7c0cf46423ea5f728c41dfbc/src/renderers/shaders/ShaderChunk/colorspace_pars_fragment.glsl.js#L23-L25)). Is it something we should add there, as well?

That aside - is it important that the values be clamped to in-gamut after this conversion? It's very likely for the Linear sRGB values to come into this function with values far greater that 1.0 and some effects such as screen space bloom depend on those larger-than 1 values being output to the render target.

Reply:
> This isn't done in our other color conversion functions (see P3 to Linear sRGB conversion). Is it something we should add there, as well?

I prefer to think of the color conversion functions themselves as open-domain-to-open-domain. But certainly any particular invocation of the functions needs to decide whether to clamp, gamut map, or keep the full domain.

> ... some effects such as screen space bloom depend on those larger-than 1 values being output to the render target

I'm trying hard to encourage users to put HDR effects like bloom before tone-mapping. See [forum discussion](https://discourse.threejs.org/t/different-color-output-when-rendering-to-webglrendertarget/57494/7). But yes, a lot of people do this. My preference would mainly be consistency, whichever result that means. It looks like our [existing tone mappers](https://github.com/mrdoob/three.js/blob/dev/src/renderers/shaders/ShaderChunk/tonemapping_pars_fragment.glsl.js) mostly do clamp before color space conversion.